app.py

```diff
@@ -1,40 +1,40 @@
 from typing import Dict
 
-import streamlit as st
-import wikipedia
-from transformers import Pipeline
-from transformers import pipeline
+import streamlit as st  # type: ignore
+import wikipedia  # type: ignore
+from transformers import Pipeline, pipeline  # type: ignore
 
-NUM_SENT = 10
+from config import config
 
 
-@st.cache
 def get_qa_pipeline() -> Pipeline:
```

Comment:
Can we add back the `@st.cache` for the pytorch code path, since otherwise I believe it does get reloaded.

Reply:
1. I have added a conditional decorator, that takes in a function and returns a streamlit cached function if framework used is pytorch. 
2. Caching `answer_question` and `get_qa_pipeline()` functions provide performance boost.  It also avoids loading the transformer pipeline again as you mentioned.

Reply:
Looks good!